### PR TITLE
Mchiou/8082 use docker containers in workflows

### LINKED ISSUE #8082: Use docker containers in workflows

### PR DESCRIPTION
**Pull request by ttmchiou**
PR to update workflows to build docker containers and use these docker containers in our workflow

Changes include 
Updating workflows to run code in docker containers 
New workflow to build docker containers and upload as artifacts

### CODE REVIEW

**Comment by TT-billteng** [comment]
can't you merge this PR directly into main?

**Comment by ttmchiou** [comment]
will require merging #7949 PR for dockerfile and scripts

File: .github/workflows/full-regressions-and-models.yaml [hunk 1]
```diff
@@ -41,10 +47,14 @@ jobs:
...
       - name: Run frequent reg tests scripts
         timeout-minutes: 210
         run: |
-          source ${{ github.workspace }}/python_env/bin/activate
-          cd $TT_METAL_HOME
-          export PYTHONPATH=$TT_METAL_HOME
-          ./tests/scripts/run_tests.sh --tt-arch ${{ matrix.arch }} --pipeline-type frequent_${{ matrix.frequent-type }} --dispatch-mode slow
+          ./scripts/docker/run_docker_cmd.sh \
+            --device /dev/tenstorrent -- \
+            /bin/bash -c "source python_env/bin/activate \
```
Comment (TT-billteng) [inline .github/workflows/full-regressions-and-models.yaml:56]: why are we activating twice here?

File: .github/workflows/run-profiler-regression.yaml [hunk 2]
```diff
@@ -18,6 +72,8 @@ jobs:
           {arch: wormhole_b0, runs-on: ["wormhole_b0", "multi-chip-num-pcie-1", "multi-chip-num-chips-1"]},
           # N300
           {arch: wormhole_b0, runs-on: ["wormhole_b0", "multi-chip-num-pcie-1", "multi-chip-num-chips-2"]},
+          # N300 2x4
+          {name: "n300-2x4", arch: wormhole_b0, runs-on: ["wormhole_b0", "multi-chip-num-pcie-4", "multi-chip-num-chips-8"]},
```
Comment (TT-billteng) [inline .github/workflows/run-profiler-regression.yaml:76]: is this a merge conflict that didn't get resolved properly?

File: .github/workflows/tgg-unit-tests.yaml [hunk 3]
```diff
@@ -42,10 +48,11 @@ jobs:
...
+      - uses: ./.github/actions/load-docker-image
       - name: Run unit regression tests
         timeout-minutes: 45
         run: |
-          source ${{ github.workspace }}/python_env/bin/activate
-          cd $TT_METAL_HOME
-          export PYTHONPATH=$TT_METAL_HOME
-          ${{ matrix.test-group.cmd }}
+          ./scripts/docker/run_docker_cmd.sh --device /dev/tenstorrent -- \
+          /bin/bash -c "source python_env/bin/activate \
```
Comment (TT-billteng) [inline .github/workflows/tgg-unit-tests.yaml:56]: why are we activating environment twice?

**Pull request closed without merging**